### DISCUSSION TITLE
Overriding css variable

### DISCUSSION THREAD

**Discussion by hansyulian** [mantinedev/mantine#6015] [Q&A]
### Reference
https://github.com/orgs/mantinedev/discussions/4418

### TLDR
I must either:
- I must put `!important` which is not nice
- I must define in `style` inline which not sure if it's proper

### Problem
So i'm creating a custom grid for gallery that have a very tight gutter, the images inside will have height defined by the calculating some base and gutter size. here is the scss styling:

```
.hero-container {
  height: var(--hero-image-size);
}

.sub3-image-container {
  height: calc((var(--hero-image-size) - 2 * var(--image-spacing)) / 3);
}

.sub2-image-container {
  height: calc((var(--hero-image-size) - var(--image-spacing)) / 2);
}

.container {
  // later adjust the sizing for responsive here
  // and the subs will auto recalculate
  // synchronize this value with the gutter
  --image-spacing: rem(2);
  --grid-gutter: var(--image-spacing); // put !important here to achieve the TLDR #1
  --hero-image-size: rem(580);
  @include larger-than($mantine-breakpoint-sm) {
    --hero-image-size: rem(580);
  }
  @include larger-than($mantine-breakpoint-md) {
    --hero-image-size: rem(580);
  }
}
```

and this is the tsx:
```
<Grid 
    className={classes.container} 
    style={{ "--grid-gutter": rem(2) }} // this is the current working as TLDR #2
>
      <Grid.Col span={{ md: 7 }}>
        <GalleryImage image={heroImage} type="hero" />
      </Grid.Col>
      <Grid.Col span={{ md: 5 }}>
        <Grid 
                className={classes.container} 
                style={{ "--grid-gutter": rem(2) }} // this is the redundant declaration as the gutter get override here
>
          {subImagesDisplay.map((image, index) => (
            <Grid.Col span={6} key={`gallery-image-${index}`}>
              <GalleryImage image={image} type="sub3" />
            </Grid.Col>
          ))}
        </Grid>
      </Grid.Col>
    </Grid>
```

So currently i am able to achieve, but doesn't feels like the solution is a proper one. I think the best solution is to be still defined in the `.container` class. Need advice thx

**Comment by rtivital** [comment]
These variables are meant to be overridden by component props, not with CSS. If you want to use them in CSS, it is required to use important.

**Reply by hansyulian** [reply]
ok, thanks.

Meanwhile, may I know what is the rationale and the use case to the point that we don't allow the css to override these variables?

**Reply by rtivital** [reply]
These variables are defined in style attribute which makes them more specific.
